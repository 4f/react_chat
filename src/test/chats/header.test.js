/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatHeader from 'components/chats/header';

jest.mock('components/ava', () => () => 'Avatar');
jest.mock('components/chats/header/chat_menu', () => () => 'ChatMenu');
jest.mock('components/chats/header/user_menu', () => () => 'UserMenu');

const mockProps = {
  user: {
    _id: 'ura',
    firstName: 'Name',
    lastName: 'Surname',
    username: 'username',
    isMember: true,
    isCreator: true,
    isInChat: true,
  },
  chat: {
    _id: '13',
    title: 'My Chat',
    messages: [
      {
        chatId: '13',
        content: 'mmm',
        sender: {},
        createdAt: '2018-03-26T11:11:11.100Z',
      },
    ],
  },
  logout: jest.fn(),
  redirect: jest.fn(),
  leaveChat: jest.fn(),
  joinChat: jest.fn(),
  deleteChat: jest.fn(),
  editUser: jest.fn(),
};

describe('<ChatHeader />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatHeader {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without active chat', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} activeChat={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
