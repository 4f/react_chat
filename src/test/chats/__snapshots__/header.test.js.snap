// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> renders correctly 1`] = `
<header
  className="MuiPaper-root-92 MuiPaper-shadow4-98 MuiAppBar-root-84 MuiAppBar-positionFixed-85 MuiAppBar-colorPrimary-90 mui-fixed ChatHeader-appBar-74"
>
  <div
    className="MuiToolbar-root-119 MuiToolbar-gutters-120"
    color="contrast"
  >
    Avatar
    <h2
      className="MuiTypography-root-121 MuiTypography-title-127 MuiTypography-alignLeft-133 ChatHeader-appBarTitleLeft-75"
    >
      ChatMenu
    </h2>
    <h2
      className="MuiTypography-root-121 MuiTypography-title-127 MuiTypography-alignLeft-133 ChatHeader-appBarTitleRight-78"
    >
      UserMenu
    </h2>
    Avatar
  </div>
</header>
`;

exports[`<ChatHeader /> renders without active chat 1`] = `
<header
  className="MuiPaper-root-92 MuiPaper-shadow4-98 MuiAppBar-root-84 MuiAppBar-positionFixed-85 MuiAppBar-colorPrimary-90 mui-fixed ChatHeader-appBar-74"
>
  <div
    className="MuiToolbar-root-119 MuiToolbar-gutters-120"
    color="contrast"
  >
    Avatar
    <h2
      className="MuiTypography-root-121 MuiTypography-title-127 MuiTypography-alignLeft-133 ChatHeader-appBarTitleLeft-75"
    >
      ChatMenu
    </h2>
    <h2
      className="MuiTypography-root-121 MuiTypography-title-127 MuiTypography-alignLeft-133 ChatHeader-appBarTitleRight-78"
    >
      UserMenu
    </h2>
    Avatar
  </div>
</header>
`;
